index.bs

```diff
@@ -318,6 +339,285 @@ NOTE: The size of the last layer can be determined by subtracting the sum of the
+
+The output reconstructed image is made up of the output samples, whose values are each clamped to fit in the number of bits per sample as defined by the 'pixi' property of the reconstructed image item. The field <code>full_range_flag</code> of the 'colr' property of <code>colour_type</code> 'nclx' also defines a range of values to clamp to, as defined in [[!CICP]].
+
+<div class="example">A primary [=MIAF image item=] can be completed with another hidden [=MIAF image item=], both used as input image items to a [=Sample Transform Derived Image Item=], with an expression corresponding to ReconstructedSample = 256 × PrimarySample + HiddenSample. The primary [=MIAF image item=] and the [=Sample Transform Derived Image Item=] are both part of the same [=altr=] group. This will be perceived as a backward-compatible regular 8-bit image to readers not supporting [=Sample Transform Derived Image Items=], and can be decoded as a 16-bit image otherwise, making that pattern a bit-depth extension mechanism.</div>
+
+<h5 id="sample-transform-syntax">Syntax</h5>
+
+An expression is a series of tokens. A token is an operand or an operator. An operand can be a literal constant value or a sample value. A stack is used to keep track of the results of the subexpressions. An operator takes either one or two input operands. Each unary operator pops one value from the stack. Each binary operator pops two values from the stack, being the right operand then the left operand. Each token results in a value pushed to the stack. The single remaining value in the stack after evaluating the whole expression is the resulting output sample.
+
+```c
+aligned(8) class SampleTransform {
+    unsigned int(2) version = 0;
+    unsigned int(4) reserved;
+    unsigned int(2) bit_depth; // Signed 8, 16, 32 or 64-bit.
+    unsigned int(8) token_count;
+    for (i=0; i<token_count; i++) {
+        unsigned int(8) token;
+        if (token == 0) {
+            // Push the 'constant' value to the stack.
+            signed int(1<<(bit_depth+3)) constant;
+        } else if (token == 1) {
+            // Push the sample value from the 'input_image_item_index'th
+            // input image item to the stack. At most 'reference_count'.
+            unsigned int(8) input_image_item_index; // 1-based
+        } else {
+            if (token == 2 || token == 3 || token == 11 || token == 12) {
+                // Unary operator. Pop an operand from the stack.
+            } else {
+                // Binary operator. Pop a right operand
+                // then a left operand from the stack.
+            }
+            // Apply operator 'token' and push the result to the stack.
+        }
+    }
+}
+```
+
+<h5 id="sample-transform-semantics">Semantics</h5>
+
+<dfn noexport>version</dfn> shall be equal to 0.
```

Comment:
noexport is not strictly necessary since a pure `<dfn>` does not export by default. But I don't mind being explicit.

Reply:
I did not follow closely what these tags were for. I mostly copy-pasted stuff.